artemis/remote/nanny.py

```diff
@@ -0,0 +1,224 @@
+import atexit
+import signal
+import sys
+import threading
+import time
+
+import os
+
+from artemis.experiments.experiment_record import get_current_experiment_name, get_current_experiment_dir
+
+
+class ManagedChildProcess(object):
+    def __init__(self,cp,monitor_for_termination,monitor_if_stuck_timeout):
+        self.cp = cp
+        self.monitor_for_termination = monitor_for_termination
+        self.monitor_if_stuck_timeout = monitor_if_stuck_timeout
+
+    def get_process(self):
+        return self.cp
+
+    @property
+    def name(self):
+        return self.cp.name
+
+    def monitor_if_stuck(self):
+        return self.monitor_if_stuck_timeout is not None
+
+    def kill(self, signal=signal.SIGINT):
+        return self.cp.kill(signal)
+
+    def is_alive(self):
+        return self.cp.is_alive()
+
+    def get_name(self):
+        return self.cp.get_name()
+
+    def get_ip(self):
+        return self.cp.get_ip()
+
+    def get_id(self):
+        return self.cp.get_id()
+
+    def execute(self):
+        return self.cp.execute_child_process()
+
+    def get_id(self):
+        return self.cp.get_id()
+
+    def deconstruct(self, signum=signal.SIGINT, system_signal=False):
+        return self.cp.deconstruct(signum)
+
+class Nanny(object):
+    '''
+    Manages child processes. This class manages the start, live and deconstruction of child processes across different machines.
+    '''
+    def __init__(self):
+        self.managed_child_processes = {}
+        self.stdout_threads = {}
+        # self.original_sigint_handler = signal.getsignal(signal.SIGINT)
+        # self.original_sigterm_handler = signal.getsignal(signal.SIGTERM)
+        # signal.signal(signal.SIGINT, lambda signum, stack :self.deconstruct(signal.SIGINT,system_signal=True))
+        # signal.signal(signal.SIGTERM, lambda signum, stack :self.deconstruct(signal.SIGTERM,system_signal=True))
```

Comment:
lets remove commented code unless kept for a reason, in which case explain reason